src/runtime/setup-session.js

```diff
@@ -1,9 +1,16 @@
 import Session from './session';
+import { warn } from '../shared/log';
 
 export default function setupSession(global) {
-
   // The name of the property encodes the session/node compatibilty version
   if (!global._heimdall_session_3) {
     global._heimdall_session_3 = new Session();
   }
+
+  for (let i = 1; i < 3; i++) {
```

Comment:
Basic approach seems fine although TBH i think it would be clearer to just unroll the loop.  We really don't plan on bumping the session version often.

```js
if ('_heimdall_session_1' in global || '_heimdall_session_2' in global) {
  //warn
}
```

Also tests.